File: src/main/java/com/example/oauth/model/oauth/OAuthProvider.java
package com.example.oauth.model.oauth;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum OAuthProvider {
    KAKAO("kakao"),
    NAVER("naver"),
    TWITTER("twitter");

    private final String name;

    public static OAuthProvider nameOf(String name) {
        for (OAuthProvider provider : values()) {
            if (name.equals(provider.getName())) {
                return provider;
            }
        }
        return null;
    }
}

File: src/main/java/com/example/oauth/model/oauth/OAuthProvider.kt
package com.example.oauth.model.oauth

enum class OAuthProvider(val id: String) {
    KAKAO("kakao"), NAVER("naver"), TWITTER("twitter");

    companion object {
        fun idOf(id: String): OAuthProvider? {
            for (provider in values()) {
                if (id == provider.id) {
                    return provider
                }
            }
            return null
        }
    }

}